### ISSUE TITLE
Invalid usage of parent selector `&` not detected

### ISSUE THREAD

**Issue by cedric-anne** [stylelint-scss/stylelint-scss#568]
Accordins to [SASS documentation](https://sass-lang.com/documentation/style-rules/parent-selector)
> Because the parent selector could be replaced by a type selector like `h1`, it’s only allowed at the beginning of compound selectors where a type selector would also be allowed. For example, `span&` is not allowed.

So, IMHO, following SCSS file should trigger an error.

```scss
.container {
    .child {
        .container-hidden& {
            // someone tries to generate following rule: `.container-hidden.container .child`
        }
    }
}
```

**Comment by kristerkari** [comment]
@cedric-anne Thanks for your suggestion! However, it's the Sass compiler's job to catch and report syntax errors. The linter is meant to enforce good coding practices and code styling, etc.
